staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -338,9 +348,11 @@ func (r *crdHandler) removeDeadStorage() {
 			}
 		}
 		if !found {
-			glog.V(4).Infof("Removing dead CRD storage for %v", s.requestScope.Resource)
-			// destroy only the main storage. Those for the subresources share cacher and etcd clients.
-			s.storage.CustomResource.DestroyFunc()
+			for version, storage := range s.storages {
+				glog.V(4).Infof("Removing dead CRD storage for %v", s.requestScopes[version].Resource)
```

Comment:
move log command outside of the loop.

Reply:
We don't need the list of versions here I guess. Just print out the gk or gr is enough.

Reply:
Could be a simple fix, but lets do this in one of the follow ups too.